### PR TITLE
Changed notification library

### PR DESCRIPTION
**Pull request by skalickym**
- FS notification library changed to github.com/rjeczalik/notify to enable
  recursive watch

### CODE REVIEW

**Comment by skalickym** [comment]
@coufalja please check

File: pkg/sync/sync.go [hunk 1]
```diff
@@ -1,93 +1,95 @@
...
-func (s *syncer) uploadFile(fileDir string) {
+func (s *syncer) uploadFile(fileDir string, deleteFile bool) {
 	// Upload
 	err := s.addFileToS3(s.s3session, fileDir)
 	if err != nil {
 		log.Error(err)
+	} else {
+		if deleteFile {
+			var errDelete = os.Remove(fileDir)
+			if os.IsNotExist(errDelete) {
```
Comment (coufalja) [inline pkg/sync/sync.go]: I would say that cases opposite to this are even more important to log

File: pkg/fswatch/fswatch.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
+package fswatch
+
+import (
+	"log"
+
+	"github.com/rjeczalik/notify"
+)
+
+//Watch
+type FSWatcher struct {
```
[the remark this answers is not in the document]
Comment (skalickym) [reply pkg/fswatch/fswatch.go]: done

File: pkg/sync/sync.go [hunk 3]
```diff
@@ -1,93 +1,95 @@
...
-					log.Infof("uploading new file '%s' to S3", event.Name)
-					go s.uploadFile(event.Name)
-				default:
-					log.Debug("no changes")
-				}
-			case err := <-s.watcher.Errors:
-				log.Error(err)
-			case <-s.stop:
-				s.watcher.Close()
+	for {
```
[the remark this answers is not in the document]
Comment (skalickym) [reply pkg/sync/sync.go]: it's back! hurray!

File: pkg/sync/sync.go (hunk 1 above)
Comment (skalickym) [reply pkg/sync/sync.go]: I've removed deleting part at the end. I showed to be more tricky than expected.

File: pkg/sync/sync.go [hunk 4]
```diff
@@ -1,93 +1,95 @@
...
 
-func NewSyncer(folderToWatch, s3Region, s3Path string, stop chan struct{}) (Syncer, error) {
-	s3Path =  strings.Replace(s3Path, "s3://", "", -1)
-	watcher, err := fsnotify.NewWatcher()
-	if err != nil {
-		return nil, err
-	}
+func NewSyncer(fchan chan notify.EventInfo, folderToWatch, s3Region, s3Path string, stop chan struct{}, deleteFile bool) (Syncer, error) {
+	s3Path = strings.Replace(s3Path, "s3://", "", -1)
+	// Make the channel buffered to ensure no event is dropped. Notify will drop
```
[the remark this answers is not in the document]
Comment (skalickym) [reply pkg/sync/sync.go]: right, removed.

File: cmd/root.go [hunk 5]
```diff
@@ -25,7 +27,13 @@ var rootCmd = &cobra.Command{
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-		stop := make(chan struct{})
+		stop := make(chan notify.EventInfo)
+
+		if err := notify.Watch(folderToWatch+"/...", stop, notify.Create, notify.Write); err != nil {
+			log.Fatal(err)
```
[the remark this answers is not in the document]
Comment (skalickym) [reply cmd/root.go]: done.

**Comment by skalickym** [comment]
@coufalja could you please merge? I have no write access.

**Pull request merged**